browser/lib/util/crypto.js

```diff
@@ -14,7 +14,7 @@ var Crypto = (function() {
 	 * @param callback
 	 */
 	var generateRandom;
-	if(typeof Uint32Array !== 'undefined' && Platform.getRandomValues) {
+	if(typeof Uint32Array !== 'undefined' && Platform.getRandomValues && window.crypto) {
```

Comment:
This is the sort of check that needs to be encapsulated in the Platform object. As it stands, the effect of this change will be to force weak random numbers for nativescript, react-native, and IE11 (which uses `window.mscrypto` rather than `window.crypto`), all of which do support strong random numbers via `Platform.getRandomValues`.

To put it another way, the code already checks that `Platform.getRandomValues` is defined; the bug is that it shouldn't be defined if in a browser context where you have neither `window.crypto` nor `window.mscrypto`.